internal/store/group_test.go

```diff
@@ -1,22 +1,94 @@
 package store
 
 import (
+	corev1 "k8s.io/api/core/v1"
+	k8sResource "k8s.io/apimachinery/pkg/api/resource"
+
 	"testing"
 	"time"
 
-	"github.com/mattermost/mattermost-cloud/model"
 	"github.com/mattermost/mattermost-cloud/internal/testlib"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+func compareGroupLists(t *testing.T, list1 []*model.Group, list2 []*model.Group) {
+	for _, g1 := range list1 {
+		found := false
+		for _, g2 := range list2 {
+			if g1.ID == g2.ID {
+				found = true
+				compareGroups(t, g1, g2)
+				break
+			}
+		}
+		if !found {
+			t.Fatal("groups of lists did not match", list1, list2)
+			continue
+		}
+	}
+}
```

Comment:
As mentioned above, tests should pass without the need for this logic. Comparing manually means we will have to make sure we don't forgot to check new values as they are added to groups.

Reply:
I don't this this is right, `assert.Equal` insists that pointer addresses are identical, which obviously we get new addresses when we write an object to the database and then read it out into a new object. 

I recognize the drawbacks of checking each value manually, but I wasn't able to see with our testing library a way to indicate that pointers should be dereferenced and values compared in a generic way, short of using reflection.